diesel/src/connection/transaction_manager.rs

```diff
@@ -0,0 +1,92 @@
+use backend::UsesAnsiSavepointSyntax;
+use connection::Connection;
+use result::QueryResult;
+
+/// Manages the internal transaction state for a connection. You should not
+/// interface with this trait unless you are implementing a new connection
+/// adapter. You should use [`Connection::transaction`][transaction],
+/// [`Connection::test_transaction`][test_transaction], or
+/// [`Connection::begin_test_transaction`][begin_test_transaction] instead.
```

Comment:
I'm going to do it as a separate commit. I need to rebase something off this branch to open a PR